### PR TITLE
tests(ci): add basic CI tests and fix POST requests bug

### PR DESCRIPTION
**Pull request by Shillaker**
## Summary

**_What's changed?_**

- Split CI build step (running on 16.x and 18.x) into separate workflow
- Add CI step to test all examples on 16 and 18. Successful run [here](https://github.com/scaleway/serverless-functions-node/actions/runs/6800402224)
- Add CI unit test step (running on 18.x only;16/18 compatibility tested in examples). Successful run [here](https://github.com/scaleway/serverless-functions-node/actions/runs/6800402224/job/18488856017)
- Add simple GET/POST request unit tests
- Do not parse JSON inputs as an object (fix for https://github.com/scaleway/serverless-functions-node/issues/64)
- Add dev docs to help set up project locally
- Split HTTP GET and POST examples

**_Why do we need this?_**

- Currently no CI tests
- Currently not checking examples work
- The Scaleway function Node runtime does not take parsed JSON inputs (and currently POST requests break)

**_How have you tested it?_**

New CI tests.

## Checklist

- [x] I have reviewed this myself
- [x] There is a unit test covering every change in this PR
- [x] I have updated the relevant documentation

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -6,7 +6,8 @@
   "types": "framework/dist/local/index.d.ts",
```
Comment (cyclimse) [inline package.json:6]: not sure if you wanted to do it in a separate MR, but we'll need to bump the version to do a release

**Review by Bemilie — DISMISSED** [review]
It looks greats, thanks for adding the tests with the GHA 👌

File: tests/requests.test.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    await server.ready();
+  });
+
+  test("GET request response", async () => {
+    // Make request
+    const response = await fetch("http://localhost:8080", {
+      method: "GET",
+    });
+
+    var responseText = await response.text();
```
Comment (cyclimse) [inline tests/requests.test.ts:26]:
```suggestion
    const responseText = await response.text();
```

File: tests/requests.test.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    let data = { foo: "bar" };
+    const response = await fetch("http://localhost:8080", {
+      headers: {
+        "Content-Type": "application/json",
+      },
+      method: "POST",
+      body: JSON.stringify(data),
+    });
+
+    var responseText = await response.text();
```
Comment (cyclimse) [inline tests/requests.test.ts:41]:
```suggestion
    const responseText = await response.text();
```

File: examples/with_http_get/handler.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+        res.on("end", () => {
+          resolve({
+            statusCode: 200,
+            headers: { "Content-Type": ["application/json"] },
+            body: JSON.stringify(JSON.parse(dataString), null, 4),
+          });
+        });
+      },
+    );
```
Comment (cyclimse) [inline examples/with_http_get/handler.js:24]: I don't think you need to fix it, but IMO mixing callbacks and promises is confusing.

**Review by cyclimse — DISMISSED** [review]
LGTM, just a few nits. Don't forget to bump the version :smile:! IMO, this can be a minor semantic version bump since it's a major fix to the framework.

File: .github/workflows/tests.yml [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+  test:
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: "Set up node 18"
+        id: setup-node
+        uses: actions/setup-node@v4
+        with:
+          node-version: "18.x"
```
[the remark this answers is not in the document]
Comment (Shillaker) [reply .github/workflows/tests.yml:20]: For compatibility checks I will add a process to run through all the examples.

Once we have this, I think we can just run the unit/integration tests here with 18. It becomes annoying to have to support the old syntax of 16 (e.g. there's no `fetch` in 16).

File: examples/with_http_get/handler.js (hunk 4 above)
Comment (Shillaker) [reply examples/with_http_get/handler.js:24]: Yes I agree, but this code was already here so out of scope for this PR probably

**Review by Bemilie — APPROVED** [review]

**Review by cyclimse — APPROVED** [review]
Great job for the pipeline to run the examples!
